### PR TITLE
Feature Filled Entry with narrow underline 

### PR DESCRIPTION
**Pull request by fpabl0**
### Description:

Background color and narrow underline for widget.Entry to be more inline with Material Design Specs.

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [ ] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

File: theme/theme.go
Comment (andydotxyz) [inline theme/theme.go:176]: This was a good question - there is a PR open to fix this already ;)

File: theme/theme.go
Comment (andydotxyz) [inline theme/theme.go:134]: I could not think of a more generic way to describe this. Perhaps someone else can?
Or would we be better off hard coding it in Entry as `Padding()/2` like some other widgets do?

File: theme/theme.go
Comment (andydotxyz) [inline theme/theme.go:104]: We try to go for the generic naming from Material, rather than being widget specific.
I think on the call yesterday was saw that MD uses FocusColor for entry background, so my vote gets that.
We already defined this name in some areas, but it was not fully, or correctly, implemented. So we can fix that here.

File: widget/entry.go
Comment (andydotxyz) [inline widget/entry.go:1073]: :)
I'm happy with this approach - but should the line perhaps come in on the left/right in the same manner?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Thanks for this. A small discussion about naming but it's probably just the thing.

Please note that there are test failures now that need to be addressed.
When we change graphical elements some rendering test will fail, it outputs the new version in a "failed" folder and prints the path - you just drag that over the old image and commit :). GitHub will then allow us to see the change inline.

Run `go test ./...` at the project root to see the list of issues.

**Comment by fpabl0** [comment]
@andydotxyz 
> Thanks for this. A small discussion about naming but it's probably just the thing.

You're welcome.

> When we change graphical elements some rendering test will fail, it outputs the new version in a "failed" folder and prints the path. 

I have just done that (I have moved images from driver/software/testdata/failed to driver/software/testdata/, is it right?). However, they are still tests failed. And not only for entry, for widget.Form too (I think because it uses Entry too?). I wanted to solve these problems, but I am not sure how these tests work (specially test.AssertRendersToMarkup(...)), is there a guide that show how to use Fyne test suite?

File: widget/entry.go
Comment (fpabl0) [reply widget/entry.go:1073]: You mean substracting "r.box.StrokeWidth/2" in "line" too, right?

**Comment by andydotxyz** [comment]
> specially test.AssertRendersToMarkup(...)

Oh I am sorry I forgot about this (there is an ongoing discussion about how to make it as easy as the image file changes you did).
At this time you need to see the output of the test failure, and copy the change back into the embedded XML in the test code. Apologies for this. There is an open ticket to make it easier that may land tomorrow... #1669

**Comment by andydotxyz** [comment]
I just realised you are probably best holding off for now. There is another big Entry PR and it will conflict with yours.
Hopefully we can land that today and you can rebase.
Otherwise the merge conflict on the test files will be huge.
Sorry I forgot about that before! @fpabl0

**Comment by fpabl0** [comment]
@andydotxyz 
> At this time you need to see the output of the test failure, and copy the change back into the embedded XML in the test code.

Ok, no problems.

> There is another big Entry PR and it will conflict with yours.
Hopefully we can land that today and you can rebase.

Oh, it's ok, I will wait then. Could you tag that PR, just to be aware when it is possible to rebase this?

File: theme/theme.go
Comment (fpabl0) [reply theme/theme.go:104]: Oh yes, I forgot about it. However the obvious first step must be update the FocusColor with right value, isn't it? Otherwise, the entry will not have any form (just a blue rectangle).

**Comment by andydotxyz** [comment]
Apologies I forgot to ping. All the pending Entry and Scroll changes that I am aware about are now landed.

Also, we prefer people not to use force-push once a review has been added, because it makes incremental reviewing very difficult.

File: theme/theme.go
Comment (andydotxyz) [reply theme/theme.go:188]: This fix was landed

File: widget/entry.go
Comment (andydotxyz) [reply widget/entry.go:1044]: I imagined this would be required for the line and background to align, yes. Just on the x/width dimension.

File: theme/theme.go
Comment (fpabl0) [reply theme/theme.go:144]: I don't like the idea of hardcoding the size with padding as those are completely different sizes (they are not related).

File: widget/entry.go
Comment (fpabl0) [reply widget/entry.go:1044]: Ok, I will do that change

**Comment by fpabl0** [comment]
> Apologies I forgot to ping. All the pending Entry and Scroll changes that I am aware about are now landed.

No problems, great.

> Also, we prefer people not to use force-push once a review has been added, because it makes incremental reviewing very difficult.

Sorry about that, I don't know how to do it without forcing, I was following this [guide](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request). Could you guide how to do without `push force` or a link that can guide me? Thanks, and sorry for inconveniences.

**Comment by andydotxyz** [comment]
> I don't know how to do it without forcing,

Actually if you are rebasing there is probably no way to not force. Once the reviews have started then `git merge develop` allows you to preserve history.
It's worth noting that once conflicts arise a rebase may not even be possible.
Some times it's just easier to open a new PR :(

**Comment by fpabl0** [comment]
@andydotxyz 
> Not really, it just I don't know if the new value for FocusColor is ok. I am using a color a bit darker than current hover color, is that ok?

I see, that's ok. I will close this then.

**Comment by andydotxyz** [comment]
FocusColor now merged, so this work should be unblocked.

**Pull request closed without merging**
